js/views/navigation.js

```diff
@@ -46,6 +51,11 @@ define(function(require) {
 				this.$el.hide();
 				$('#app-navigation-toggle').css('background-image', 'none');
 			}
+		},
+		_adjustButtonWidth: function() {
+			var newMailButtonContainer = $('#mail-new-message-fixed');
+			var innerWidth = $('#app-navigation-accounts').width();
+			newMailButtonContainer.attr('style', 'width: ' + innerWidth + 'px !important');
```

Comment:
Dammit, I shoud I kept my mouth shut! :p 
I will check on this later on the weekend!